### PR TITLE
solarcity: parametrize logs and make other stuff optional

### PR DESCRIPTION
**Pull request by 7AC**
This unblocks the following command

```
DEBUG_MODE=1 SOLARCITY_PASSWORD=xxx SOLARCITY_USER=xx@xx.xx LOGFILE=logs/solarcity.log DARKSKY_API_KEY=xxx ./solarcity.py --report --no_email --no_tweet
```

I do still get an empty report though so I wonder if something is already broken:

```
Would email message:
Hi there, below is the weekly SolarCity generation report:

Total production this week: 0.0 kWh
Total production in the last 30 days: 0.0 kWh
Total production in the last 365 days: 0.0 kWh

Lifetime generation is 0.0 kWh.

Regards,
Teslaliving
http://teslaliving.net
```

### CODE REVIEW

**Comment by the-mace** [comment]
Does the DATA_FILE have contents? ('solarcity.json') and is it in the directory you're running in? Without that and without contents the outputs would be all zero. 

(its running fine for me in my environment so this is likely a local ENV type difference again as in the changes above)

**Comment by 7AC** [comment]
@the-mace thanks I don't. What do I need to put in it?

**Comment by the-mace** [comment]
> @the-mace thanks I don't. What do I need to put in it?

Nothing, if you do a successful poll it will create a "solarcity.json" file for you. I just ran with no existing file and saw one bug which I fixed and pushed up. With that fix it will poll and leave a working file.

This is my complete file after a run from no file:
`{"config": {"lastdailytweet": "20190519"}, "data": {"20190519": {"daylight": 14.7, "cloud": 36, "production": 40.8}}}`

**Comment by 7AC** [comment]
Thank you. I retried with your commit but I still get all zeros with the above command

**Comment by the-mace** [comment]
> Thank you. I retried with your commit but I still get all zeros with the above command

There wont be any data until you run a --daily poll, did you do that?

**Comment by 7AC** [comment]
Thank you that worked after a couple more tweaks that I added to the PR: https://github.com/the-mace/evtools/pull/3/commits/0bb9f7a3d5455e9e7dcecd94950bfc99842ad9ab

File: solarcity.py [hunk 1]
```diff
@@ -587,11 +596,13 @@ def main():
...
                 special = "low"
-            tweet_production(daylight_hours, cloud_cover, production, special)
-            data['config']['lastdailytweet'] = current_day
+            if not args.no_tweet:
+                tweet_production(daylight_hours, cloud_cover, production, special)
+                data['config']['lastdailytweet'] = current_day
             data_changed = True
-            # Now upload to pvoutput
-            upload_to_pvoutput(data, current_day)
+            if args.pvoutput is not None:
```
Comment (the-mace) [inline solarcity.py:588]: This changes the behavior. If you provide "--pvoutput" a parameter it ONLY sends output to pvoutput. If you do daily it does the daily stuff and the output to pvoutput. Could re-order the parsing to catch daily first or solve it another way.

File: solarcity.py [hunk 2]
```diff
@@ -79,7 +79,9 @@
 AUTH_URL = 'https://login.solarcity.com/account/SignIn'
 
 # Where logging output from this tool goes. Modify path as needed
-LOGFILE = os.path.expanduser("~/script/logs/solarcity.log")
+LOGFILE = os.environ.get('LOGFILE')
```
Comment (the-mace) [inline solarcity.py]: Probably could use a better name here. SOLAR_LOGFILE maybe. LOGFILE could pick up something else.

**Review by the-mace — CHANGES_REQUESTED** [review]

**Comment by the-mace** [comment]
Merged it but Tesla took SolarCity page offline so this whole script wont work any more :(

**Comment by stephenjamieson** [comment]
https://solarguard.solarcity.com/Kiosk/Login/Default.aspx dunno why this site is still around (probably abandoned) but check it out....  Otherwise check out https://github.com/jbuehl/solaredge

**Pull request merged**